gateway-ha/src/main/java/io/trino/gateway/baseapp/BaseApp.java

```diff
@@ -159,68 +107,58 @@ protected List<Module> addModules(HaGatewayConfiguration configuration)
         return modules;
     }
 
-    /**
-     * Supply a list of managed apps.
-     */
-    protected List<Managed> addManagedApps(
-            HaGatewayConfiguration configuration, Environment environment, Injector injector)
+    @Override
+    public void configure(Binder binder)
+    {
+        registerWithInjector(this.haGatewayConfiguration, binder);
+    }
+
+    private static void registerWithInjector(HaGatewayConfiguration configuration, Binder binder)
+    {
+        registerAuthFilters(binder);
+        registerProviders(binder);
+        addManagedApps(configuration, binder);
+        registerResources(binder);
+    }
+
+    private static void addManagedApps(HaGatewayConfiguration configuration, Binder binder)
     {
-        List<Managed> managedApps = new ArrayList<>();
         if (configuration.getManagedApps() == null) {
             logger.error("No managed apps found");
-            return managedApps;
+            return;
         }
-        configuration
-                .getManagedApps()
-                .forEach(
-                        clazz -> {
-                            try {
-                                Class c = Class.forName(clazz);
-                                LifecycleEnvironment lifecycle = environment.lifecycle();
-                                lifecycle.manage((Managed) injector.getInstance(c));
-                                logger.info("op=register type=managed item=%s", c);
-                            }
-                            catch (Exception e) {
-                                logger.error(e, "Error loading managed app");
-                            }
-                        });
-        return managedApps;
+        configuration.getManagedApps().forEach(
```

Comment:
Do we need it? In the configuration classes these are always:
``managedApps:
  - io.trino.gateway.ha.GatewayManagedApp
  - io.trino.gateway.ha.clustermonitor.ActiveClusterMonitor
``

We can bind GatewayManagedApp.class and ActiveClusterMonitor.class

Reply:
I agree we should use static binding, but that changes will also affect `modules` and others.
I'd like to remove `modules` and `managedApps` in the configuration in the future.
We'll need to update the design of configuration and that requires further discussion.

Since these are nice to have changes and not a must for the migration to Airlift, I'd like to leave it as is.